collectd/src/main/java/com/microsoft/applicationinsights/collectd/ApplicationInsightsWriter.java

```diff
@@ -0,0 +1,227 @@
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.collectd;
+
+import java.util.HashMap;
+import java.util.List;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.collectd.internal.ApplicationInsightsWriterLogger;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.util.LocalStringsUtils;
+import com.microsoft.applicationinsights.telemetry.MetricTelemetry;
+import org.collectd.api.*;
+
+/**
+ * Created by yonisha on 4/26/2015.
+ */
+public class ApplicationInsightsWriter implements CollectdConfigInterface,
+        CollectdInitInterface,
+        CollectdWriteInterface,
+        CollectdShutdownInterface {
+
+    // region Consts
+
+    private static final String PLUGIN_NAME = "ApplicationInsightsWriter";
+    private static final String INSTRUMENTATION_KEY_CONFIGURATION_KEY = "InstrumentationKey";
+    private static final String SDK_LOGGER_CONFIGURATION_KEY = "SDKLogger";
+
+    // endregion Consts
+
+    // region Members
+
+    private TelemetryConfiguration telemetryConfiguration;
+    private TelemetryClient telemetryClient;
+    private ApplicationInsightsWriterLogger logger = null;
+
+    // endregion Members
+
+    // region Ctors
+
+    public ApplicationInsightsWriter() {
+        Collectd.registerConfig(PLUGIN_NAME, this);
+        Collectd.registerInit(PLUGIN_NAME, this);
+        Collectd.registerWrite(PLUGIN_NAME, this);
+        Collectd.registerShutdown(PLUGIN_NAME, this);
+
+        this.logger = new ApplicationInsightsWriterLogger();
+        this.telemetryConfiguration = TelemetryConfiguration.getActive();
+    }
+
+    protected ApplicationInsightsWriter(TelemetryClient telemetryClient, ApplicationInsightsWriterLogger logger) {
+        this.telemetryClient = telemetryClient;
+        this.telemetryConfiguration = TelemetryConfiguration.getActive();
+        this.logger = logger;
+    }
+
+    // endregion Ctors
+
+    // region Public
+
+    /**
+     * Reads the configuration provided for the writer plugin.
+     * @param configuration The configuration.
+     * @return Zero on success, false otherwise.
+     */
+    public int config(OConfigItem configuration) {
+        logger.logInfo("Loading configuration.");
+        logger.logDebug("Configuration found: " + configuration);
+
+        List<OConfigItem> children = configuration.getChildren();
+        if (children.size() < 1) {
+            logger.logError("Configuration error: application instrumentation key must be set.");
+
+            return 1;
```

Comment:
What's the meaning of 1?
Please use constants with meaningful names (also relevant for the return values below)